### ISSUE TITLE
Support DTLS resume session

### ISSUE THREAD

**Issue by AndreasHuber-CH** [bcgit/bc-java#467]
The "bctls" library has an API to support session ressumption.
It looked like we could implement the method "getSessionToResume(byte[] sessionID)" in our implementation of org.bouncycastle.tls.TlsServer.
But when I looked closer at the BC code I found out that this method is never called and the feature to resume a session is not implemented (and consequently the ServerHello never contains a session ID and any session ID present in the ClientHello is ignored).

Do you have any plans to implement this feature ?

**Comment by peterdettman** [comment]
Yes, it's been on our TODO list for quite a while now and we still plan to get to it. It's perhaps not as simple to add as it might seem since 1. it requires the abbreviated handshake be supported in the server state machine, and 2. there are many clauses from other RFCs describing different behaviour in the case of session resumption, many with security implications if they are missed or misapplied.

**Comment by peterdettman** [comment]
Added by https://github.com/bcgit/bc-java/commit/00d216a29f387f269033c2b41a67ec0940185acf .
